src/all/comicgrowl/src/eu/kanade/tachiyomi/extension/all/comicgrowl/ComicGrowl.kt

```diff
@@ -1,63 +1,216 @@
 import okhttp3.HttpUrl.Companion.toHttpUrl
-import okhttp3.OkHttpClient
 import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.nodes.Document
 import org.jsoup.nodes.Element
+import uy.kohesive.injekt.injectLazy
+import java.text.SimpleDateFormat
+import java.util.Locale
 
-// TODO: get manga status
-// TODO: filter by status
-// TODO: change cdnUrl as a array(upstream)
-class ComicGrowl : GigaViewer(
-    "コミックグロウル",
-    "https://comic-growl.com",
-    "all",
-    "https://cdn-img.comic-growl.com/public/page",
-) {
+class ComicGrowl(
+    override val lang: String = "all",
+    override val baseUrl: String = "https://comic-growl.com",
+    override val name: String = "コミックグロウル",
+    override val supportsLatest: Boolean = true,
+) : ParsedHttpSource() {
 
-    override val publisher = "BUSHIROAD WORKS"
+    override val client = super.client.newBuilder()
+        .addNetworkInterceptor(ImageDescrambler::interceptor)
+        .build()
 
-    override val chapterListMode = CHAPTER_LIST_LOCKED
+    override fun headersBuilder(): Headers.Builder {
+        return super.headersBuilder().set("Referer", "$baseUrl/")
+    }
 
-    override val supportsLatest: Boolean = true
+    override fun popularMangaRequest(page: Int) = GET("$baseUrl/ranking/manga", headers)
 
-    override val client: OkHttpClient =
-        super.client.newBuilder().addInterceptor(::imageIntercept).build()
+    override fun popularMangaNextPageSelector() = null
 
-    override fun popularMangaRequest(page: Int): Request = GET(baseUrl, headers)
+    override fun popularMangaSelector() = ".ranking-item"
 
-    // Show only ongoing works
-    override fun popularMangaSelector(): String = "ul[class=\"lineup-list ongoing\"] > li > div > a"
+    override fun popularMangaFromElement(element: Element): SManga {
+        return SManga.create().apply {
+            setUrlWithoutDomain(element.selectFirst("a")!!.absUrl("href"))
+            title = element.selectFirst(".title-text")!!.text()
+            setImageUrlFromElement(element)
+        }
+    }
 
-    override fun popularMangaFromElement(element: Element) = SManga.create().apply {
-        title = element.select("h5").text()
-        thumbnail_url = element.select("div > img").attr("data-src")
-        setUrlWithoutDomain(element.attr("href"))
+    override fun mangaDetailsParse(document: Document): SManga {
+        val infoElement = document.selectFirst(".series-h-info")!!
+        val authorElements = infoElement.select(".series-h-credit-user-item .article-text")
+        val updateDateElement = infoElement.selectFirst(".series-h-tag-label")
+        return SManga.create().apply {
+            title = infoElement.select("h1 > span")[1]!!.text()
+            author = authorElements.joinToString { it.text() }
+            description = infoElement.selectFirst(".series-h-credit-info-text-text p")?.wholeText()?.trim()
+            setImageUrlFromElement(document.getElementsByClass("series-h-img").first())
```

Comment:
```suggestion
            setImageUrlFromElement(document.selectFirst(".series-h-img"))
```

Probably need to do something about null.

Reply:
null check is in the helper function